### PR TITLE
Fix maintainer warnings when compiling with MSVC

### PR DESCRIPTION
**Pull request by nmoinvaz**
Fixes various maintainer warnings for MSVC and should help reduce them for Intel.

### CODE REVIEW

File: functable.c [hunk 1]
```diff
@@ -371,14 +371,15 @@ Z_INTERNAL uint8_t* chunkmemset_safe_stub(uint8_t *out, unsigned dist, unsigned
...
+    int32_t little_endian = sizeof(void *) == sizeof(ptrdiff_t);
 
     Assert(sizeof(uint64_t) >= sizeof(size_t),
            "crc32_z takes size_t but internally we have a uint64_t len");
     /* return a function pointer for optimized arches here after a capability test */
 
     cpu_check_features();
 
-    if (sizeof(void *) == sizeof(ptrdiff_t)) {
+    if (little_endian) {
```
Comment (Dead2) [inline functable.c]: I think little_endian is the wrong name for this variable, or at least misleading.
If i read the code correctly, this if also covers the big_endian parts.

**Comment by Dead2** [comment]
## Baseline b802a303ce8b6c86fbe3f93d59e0a82333768c0c
```
   text    data     bss     dec     hex filename
 112377    1376      32  113785   1bc79 libz-ng.so.1

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      1.339/1.366/1.377/0.010        0.867/0.888/0.900/0.009      127,294,823
 2     43.874%      1.200/1.215/1.222/0.007        0.441/0.460/0.468/0.007       48,329,104
 3     42.490%      1.279/1.289/1.296/0.005        0.369/0.383/0.389/0.006       40,118,060
 4     41.472%      1.272/1.284/1.289/0.005        0.295/0.314/0.318/0.005       32,630,860
 5     41.212%      1.371/1.388/1.396/0.007        0.300/0.308/0.317/0.005       32,425,995
 6     41.037%      1.300/1.320/1.324/0.005        0.229/0.248/0.252/0.005       25,831,161
 7     40.784%      1.262/1.272/1.275/0.003        0.174/0.185/0.187/0.003       19,253,741
 8     40.706%      1.085/1.094/1.100/0.004        0.117/0.121/0.124/0.002       12,811,162
 9     40.695%      1.203/1.216/1.221/0.004        0.107/0.122/0.127/0.005       12,807,947

 avg1  43.339%                        1.272                          0.336
 tot                                343.358                         90.850      351,502,853

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      1.342/1.367/1.375/0.007        0.874/0.887/0.894/0.005      127,294,823
 2     43.874%      1.198/1.213/1.222/0.007        0.454/0.463/0.471/0.006       48,329,104
 3     42.490%      1.280/1.291/1.296/0.004        0.376/0.387/0.393/0.005       40,118,060
 4     41.472%      1.258/1.280/1.288/0.006        0.295/0.313/0.318/0.006       32,630,860
 5     41.212%      1.375/1.387/1.393/0.005        0.297/0.309/0.317/0.007       32,425,995
 6     41.037%      1.299/1.318/1.324/0.006        0.242/0.249/0.252/0.003       25,831,161
 7     40.784%      1.265/1.272/1.278/0.004        0.177/0.184/0.187/0.003       19,253,741
 8     40.706%      1.090/1.097/1.101/0.003        0.107/0.121/0.127/0.004       12,811,162
 9     40.695%      1.204/1.217/1.221/0.004        0.110/0.119/0.124/0.004       12,807,947

 avg1  43.339%                        1.271                          0.337
 tot                                343.254                         90.978      351,502,853
```

## PR #785 a0be84cdb3de66f7603324bc5c0720f618aeb458
```
   text    data     bss     dec     hex filename
 112297    1376      32  113705   1bc29 libz-ng.so.1

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      1.345/1.364/1.374/0.008        0.868/0.882/0.892/0.006      127,294,823
 2     43.874%      1.198/1.220/1.228/0.007        0.447/0.461/0.467/0.006       48,329,104
 3     42.490%      1.271/1.295/1.301/0.006        0.373/0.388/0.393/0.005       40,118,060
 4     41.472%      1.275/1.284/1.288/0.004        0.302/0.311/0.315/0.005       32,630,860
 5     41.212%      1.381/1.391/1.396/0.004        0.293/0.313/0.317/0.006       32,425,995
 6     41.037%      1.300/1.317/1.324/0.006        0.235/0.246/0.252/0.004       25,831,161
 7     40.784%      1.262/1.272/1.275/0.003        0.171/0.183/0.187/0.004       19,253,741
 8     40.706%      1.089/1.096/1.099/0.003        0.114/0.122/0.127/0.003       12,811,162
 9     40.695%      1.202/1.215/1.222/0.005        0.111/0.121/0.127/0.004       12,807,947

 avg1  43.339%                        1.273                          0.336
 tot                                343.614                         90.840      351,502,853

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      1.355/1.372/1.381/0.007        0.862/0.882/0.895/0.009      127,294,823
 2     43.874%      1.202/1.223/1.229/0.006        0.445/0.461/0.468/0.006       48,329,104
 3     42.490%      1.283/1.294/1.299/0.004        0.372/0.385/0.393/0.006       40,118,060
 4     41.472%      1.276/1.288/1.293/0.005        0.295/0.314/0.322/0.006       32,630,860
 5     41.212%      1.367/1.385/1.393/0.006        0.303/0.311/0.317/0.004       32,425,995
 6     41.037%      1.311/1.322/1.330/0.006        0.235/0.248/0.252/0.005       25,831,161
 7     40.784%      1.268/1.276/1.279/0.003        0.174/0.184/0.191/0.004       19,253,741
 8     40.706%      1.090/1.098/1.103/0.003        0.104/0.119/0.124/0.005       12,811,162
 9     40.695%      1.207/1.221/1.227/0.005        0.114/0.121/0.124/0.003       12,807,947

 avg1  43.339%                        1.275                          0.336
 tot                                344.376                         90.786      351,502,853

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      1.342/1.369/1.378/0.008        0.868/0.883/0.892/0.007      127,294,823
 2     43.874%      1.202/1.220/1.229/0.006        0.454/0.463/0.468/0.004       48,329,104
 3     42.490%      1.289/1.297/1.303/0.004        0.370/0.386/0.393/0.007       40,118,060
 4     41.472%      1.274/1.289/1.294/0.005        0.295/0.312/0.319/0.006       32,630,860
 5     41.212%      1.377/1.392/1.400/0.007        0.300/0.311/0.317/0.004       32,425,995
 6     41.037%      1.307/1.322/1.328/0.005        0.232/0.248/0.252/0.005       25,831,161
 7     40.784%      1.257/1.274/1.280/0.005        0.171/0.183/0.188/0.005       19,253,741
 8     40.706%      1.085/1.098/1.102/0.003        0.114/0.122/0.127/0.003       12,811,162
 9     40.695%      1.206/1.218/1.223/0.004        0.107/0.122/0.127/0.005       12,807,947

 avg1  43.339%                        1.276                          0.337
 tot                                344.392                         90.893      351,502,853
```

Interestingly compiled code size goes down, so that is good. Perhaps removal of overflow handling or something, now things are cast correctly?
Output data is same size, so that looks good.

I'd like to see someone else do a benchmark of this as well, because mine are saying things got slower with this PR.
I am not sure whether that is just an artifact of code placement changes, or an actual slowdown.

**Comment by nmoinvaz** [comment]
Rebased.

**Comment by nmoinvaz** [comment]
I got similar results:

## HEAD 
```
 Tool: minigzip-head.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.006/1.020/1.027/0.005       100,930,921
 2     35.523%      1.644/1.661/1.670/0.007        75,289,402
 3     34.202%      1.978/1.990/1.997/0.005        72,490,888
 4     32.935%      2.375/2.400/2.413/0.010        69,804,533
 5     32.667%      2.549/2.568/2.579/0.008        69,236,516
 6     32.515%      2.979/2.995/3.006/0.007        68,915,050
 7     32.257%      3.976/3.994/4.005/0.009        68,367,409
 8     32.172%      6.190/6.225/6.238/0.011        68,186,790
 9     32.156%      8.623/8.672/8.687/0.016        68,154,025

 avg1  34.672%                        3.503
 tot                                945.734       661,375,534

 Tool: minigzip-head.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.015/1.023/1.030/0.004       100,930,921
 2     35.523%      1.653/1.670/1.680/0.007        75,289,402
 3     34.202%      1.981/1.996/2.005/0.006        72,490,888
 4     32.935%      2.379/2.402/2.414/0.010        69,804,533
 5     32.667%      2.559/2.577/2.587/0.007        69,236,516
 6     32.515%      2.987/3.003/3.013/0.008        68,915,050
 7     32.257%      3.978/3.998/4.015/0.011        68,367,409
 8     32.172%      6.192/6.234/6.250/0.014        68,186,790
 9     32.156%      8.651/8.684/8.701/0.014        68,154,025

 avg1  34.672%                        3.510
 tot                                947.614       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/451eece170445c51c628753f211e7719cf074197
```
 Tool: minigzip-warn.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.025/1.033/1.040/0.005       100,930,921
 2     35.523%      1.652/1.666/1.674/0.006        75,289,402
 3     34.202%      1.983/1.999/2.009/0.007        72,490,888
 4     32.935%      2.391/2.412/2.425/0.011        69,804,533
 5     32.667%      2.562/2.580/2.592/0.009        69,236,516
 6     32.515%      2.987/3.008/3.019/0.008        68,915,050
 7     32.257%      3.979/4.013/4.029/0.013        68,367,409
 8     32.172%      6.218/6.245/6.263/0.013        68,186,790
 9     32.156%      8.666/8.702/8.727/0.017        68,154,025

 avg1  34.672%                        3.518
 tot                                949.790       661,375,534
 Tool: minigzip-warn.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.028/1.034/1.039/0.003       100,930,921
 2     35.523%      1.650/1.668/1.676/0.006        75,289,402
 3     34.202%      1.986/2.002/2.011/0.007        72,490,888
 4     32.935%      2.394/2.411/2.422/0.008        69,804,533
 5     32.667%      2.572/2.584/2.592/0.006        69,236,516
 6     32.515%      3.001/3.013/3.020/0.005        68,915,050
 7     32.257%      3.995/4.016/4.024/0.008        68,367,409
 8     32.172%      6.235/6.257/6.272/0.010        68,186,790
 9     32.156%      8.655/8.704/8.720/0.016        68,154,025

 avg1  34.672%                        3.521
 tot                                950.715       661,375,534
```
I will run some tests against each commit.

**Comment by nmoinvaz** [comment]
## HEAD
```
Tool: minigzip-head.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.022/1.033/1.038/0.004       100,930,921
 2     35.523%      1.660/1.675/1.683/0.007        75,289,402
 3     34.202%      1.989/2.002/2.011/0.006        72,490,888
 4     32.935%      2.394/2.411/2.428/0.009        69,804,533
 5     32.667%      2.570/2.585/2.596/0.007        69,236,516
 6     32.515%      2.992/3.011/3.020/0.007        68,915,050
 7     32.257%      3.996/4.012/4.024/0.009        68,367,409
 8     32.172%      6.211/6.244/6.258/0.013        68,186,790
 9     32.156%      8.669/8.691/8.706/0.011        68,154,025

 avg1  34.672%                        3.518
 tot                                949.908       661,375,534

Tool: minigzip-head.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.027/1.033/1.037/0.003       100,930,921
 2     35.523%      1.661/1.675/1.685/0.007        75,289,402
 3     34.202%      1.978/2.004/2.013/0.008        72,490,888
 4     32.935%      2.378/2.412/2.431/0.014        69,804,533
 5     32.667%      2.568/2.581/2.590/0.005        69,236,516
 6     32.515%      2.986/3.013/3.025/0.010        68,915,050
 7     32.257%      3.984/4.015/4.030/0.012        68,367,409
 8     32.172%      6.233/6.254/6.266/0.009        68,186,790
 9     32.156%      8.657/8.702/8.721/0.016        68,154,025

 avg1  34.672%                        3.521
 tot                                950.635       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/4f07592a22b999004d9068c42da533e8f85dc511
```
 Tool: minigzip-warn-1.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.021/1.035/1.043/0.006       100,930,921
 2     35.523%      1.655/1.675/1.685/0.007        75,289,402
 3     34.202%      1.980/2.004/2.014/0.007        72,490,888
 4     32.935%      2.386/2.407/2.424/0.011        69,804,533
 5     32.667%      2.561/2.581/2.594/0.010        69,236,516
 6     32.515%      2.994/3.013/3.025/0.009        68,915,050
 7     32.257%      3.994/4.019/4.033/0.012        68,367,409
 8     32.172%      6.214/6.250/6.266/0.015        68,186,790
 9     32.156%      8.674/8.711/8.729/0.016        68,154,025

 avg1  34.672%                        3.522
 tot                                950.851       661,375,534

 Tool: minigzip-warn-1.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.024/1.033/1.038/0.004       100,930,921
 2     35.523%      1.661/1.671/1.680/0.006        75,289,402
 3     34.202%      1.991/2.000/2.007/0.004        72,490,888
 4     32.935%      2.391/2.405/2.419/0.008        69,804,533
 5     32.667%      2.558/2.576/2.592/0.009        69,236,516
 6     32.515%      2.993/3.007/3.017/0.006        68,915,050
 7     32.257%      3.988/4.006/4.018/0.009        68,367,409
 8     32.172%      6.209/6.242/6.250/0.009        68,186,790
 9     32.156%      8.669/8.692/8.703/0.009        68,154,025

 avg1  34.672%                        3.515
 tot                                948.981       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/4a715809f27a050eb2d702643f2967086f470b49
```
 Tool: minigzip-warn-2.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.022/1.035/1.043/0.006       100,930,921
 2     35.523%      1.658/1.674/1.683/0.006        75,289,402
 3     34.202%      1.989/2.004/2.016/0.008        72,490,888
 4     32.935%      2.393/2.412/2.425/0.010        69,804,533
 5     32.667%      2.563/2.584/2.599/0.011        69,236,516
 6     32.515%      3.003/3.015/3.025/0.006        68,915,050
 7     32.257%      3.991/4.019/4.031/0.010        68,367,409
 8     32.172%      6.232/6.250/6.261/0.008        68,186,790
 9     32.156%      8.668/8.708/8.724/0.015        68,154,025

 avg1  34.672%                        3.522
 tot                                951.025       661,375,534

Tool: minigzip-warn-2.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.025/1.033/1.039/0.004       100,930,921
 2     35.523%      1.652/1.673/1.680/0.007        75,289,402
 3     34.202%      1.986/2.001/2.014/0.007        72,490,888
 4     32.935%      2.394/2.409/2.425/0.010        69,804,533
 5     32.667%      2.569/2.585/2.595/0.008        69,236,516
 6     32.515%      2.995/3.014/3.022/0.007        68,915,050
 7     32.257%      3.988/4.018/4.030/0.011        68,367,409
 8     32.172%      6.236/6.250/6.262/0.009        68,186,790
 9     32.156%      8.678/8.705/8.716/0.010        68,154,025

 avg1  34.672%                        3.521
 tot                                950.644       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/7a0aafebcf14e6aeb729eecc222eeba9046b4a00
```
 Tool: minigzip-warn-3.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.037/1.047/1.052/0.004       100,930,921
 2     35.523%      1.651/1.669/1.676/0.006        75,289,402
 3     34.202%      1.981/2.000/2.013/0.009        72,490,888
 4     32.935%      2.372/2.404/2.415/0.010        69,804,533
 5     32.667%      2.560/2.585/2.599/0.009        69,236,516
 6     32.515%      2.997/3.011/3.022/0.007        68,915,050
 7     32.257%      3.994/4.018/4.028/0.008        68,367,409
 8     32.172%      6.208/6.257/6.269/0.012        68,186,790
 9     32.156%      8.657/8.704/8.722/0.016        68,154,025

 avg1  34.672%                        3.522
 tot                                950.825       661,375,534

 Tool: minigzip-warn-3.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.037/1.049/1.055/0.004       100,930,921
 2     35.523%      1.647/1.669/1.678/0.007        75,289,402
 3     34.202%      1.986/2.001/2.010/0.006        72,490,888
 4     32.935%      2.383/2.406/2.422/0.012        69,804,533
 5     32.667%      2.557/2.584/2.596/0.010        69,236,516
 6     32.515%      2.997/3.016/3.025/0.006        68,915,050
 7     32.257%      3.996/4.021/4.032/0.010        68,367,409
 8     32.172%      6.207/6.251/6.268/0.016        68,186,790
 9     32.156%      8.675/8.718/8.732/0.015        68,154,025

 avg1  34.672%                        3.524
 tot                                951.436       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/a8843be55c9e2b1dc4110a88f768490bb3bd6d53
```
 Tool: minigzip-warn-4.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.028/1.040/1.046/0.005       100,930,921
 2     35.523%      1.660/1.672/1.682/0.007        75,289,402
 3     34.202%      1.988/2.008/2.021/0.009        72,490,888
 4     32.935%      2.395/2.418/2.432/0.010        69,804,533
 5     32.667%      2.560/2.585/2.596/0.009        69,236,516
 6     32.515%      3.005/3.021/3.032/0.007        68,915,050
 7     32.257%      3.989/4.019/4.031/0.009        68,367,409
 8     32.172%      6.230/6.252/6.270/0.013        68,186,790
 9     32.156%      8.679/8.712/8.728/0.013        68,154,025

 avg1  34.672%                        3.525
 tot                                951.761       661,375,534

 Tool: minigzip-warn-4.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.029/1.040/1.045/0.005       100,930,921
 2     35.523%      1.651/1.674/1.687/0.009        75,289,402
 3     34.202%      1.971/2.006/2.019/0.010        72,490,888
 4     32.935%      2.395/2.419/2.431/0.009        69,804,533
 5     32.667%      2.562/2.588/2.600/0.009        69,236,516
 6     32.515%      2.995/3.021/3.031/0.008        68,915,050
 7     32.257%      3.991/4.023/4.039/0.012        68,367,409
 8     32.172%      6.235/6.263/6.284/0.014        68,186,790
 9     32.156%      8.680/8.720/8.740/0.014        68,154,025

 avg1  34.672%                        3.528
 tot                                952.620       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/98b724f9d6a4c149c6be4796052b6484df2b485d
```
 Tool: minigzip-warn-5.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.024/1.041/1.047/0.006       100,930,921
 2     35.523%      1.657/1.670/1.679/0.006        75,289,402
 3     34.202%      1.985/2.005/2.017/0.008        72,490,888
 4     32.935%      2.389/2.413/2.428/0.010        69,804,533
 5     32.667%      2.559/2.586/2.599/0.009        69,236,516
 6     32.515%      2.996/3.018/3.030/0.009        68,915,050
 7     32.257%      4.002/4.023/4.036/0.008        68,367,409
 8     32.172%      6.236/6.264/6.279/0.011        68,186,790
 9     32.156%      8.670/8.709/8.736/0.019        68,154,025

 avg1  34.672%                        3.525
 tot                                951.860       661,375,534

 Tool: minigzip-warn-5.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.028/1.038/1.044/0.004       100,930,921
 2     35.523%      1.658/1.673/1.682/0.007        75,289,402
 3     34.202%      1.990/2.005/2.016/0.007        72,490,888
 4     32.935%      2.395/2.418/2.432/0.011        69,804,533
 5     32.667%      2.570/2.586/2.598/0.008        69,236,516
 6     32.515%      3.013/3.021/3.030/0.005        68,915,050
 7     32.257%      4.007/4.026/4.038/0.010        68,367,409
 8     32.172%      6.227/6.262/6.278/0.013        68,186,790
 9     32.156%      8.685/8.716/8.735/0.017        68,154,025

 avg1  34.672%                        3.527
 tot                                952.307       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/451eece170445c51c628753f211e7719cf074197
```
 Tool: minigzip-warn-6.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.032/1.039/1.045/0.004       100,930,921
 2     35.523%      1.656/1.672/1.683/0.007        75,289,402
 3     34.202%      1.988/2.006/2.016/0.008        72,490,888
 4     32.935%      2.396/2.418/2.434/0.010        69,804,533
 5     32.667%      2.563/2.590/2.605/0.010        69,236,516
 6     32.515%      3.002/3.021/3.031/0.008        68,915,050
 7     32.257%      4.004/4.029/4.044/0.012        68,367,409
 8     32.172%      6.218/6.264/6.281/0.016        68,186,790
 9     32.156%      8.688/8.723/8.742/0.016        68,154,025

 avg1  34.672%                        3.529
 tot                                952.880       661,375,534

 Tool: minigzip-warn-6.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      1.021/1.038/1.045/0.005       100,930,921
 2     35.523%      1.649/1.674/1.686/0.008        75,289,402
 3     34.202%      1.990/2.008/2.021/0.010        72,490,888
 4     32.935%      2.397/2.418/2.430/0.009        69,804,533
 5     32.667%      2.570/2.589/2.603/0.009        69,236,516
 6     32.515%      3.000/3.021/3.029/0.007        68,915,050
 7     32.257%      3.997/4.024/4.038/0.012        68,367,409
 8     32.172%      6.227/6.270/6.284/0.013        68,186,790
 9     32.156%      8.683/8.717/8.734/0.013        68,154,025

 avg1  34.672%                        3.529
 tot                                952.739       661,375,534
```

**Comment by nmoinvaz** [comment]
It appears that maybe https://github.com/zlib-ng/zlib-ng/commit/a8843be55c9e2b1dc4110a88f768490bb3bd6d53 is the issue.

File: functable.c (hunk 1 above)
Comment (nmoinvaz) [reply functable.c]: I have renamed it.

**Comment by nmoinvaz** [comment]
## HEAD
```
 Tool: minigzip-head.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      0.998/1.009/1.014/0.004       100,930,921
 2     35.523%      1.607/1.620/1.629/0.006        75,289,402
 3     34.202%      1.929/1.947/1.959/0.008        72,490,888
 4     32.935%      2.340/2.352/2.358/0.006        69,804,533
 5     32.667%      2.514/2.528/2.538/0.008        69,236,516
 6     32.515%      2.928/2.944/2.957/0.008        68,915,050
 7     32.257%      3.897/3.922/3.933/0.009        68,367,409
 8     32.172%      6.146/6.176/6.191/0.012        68,186,790
 9     32.156%      8.609/8.678/8.707/0.027        68,154,025

 avg1  34.672%                        3.464
 tot                                935.266       661,375,534

 Tool: minigzip-head.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      0.987/1.001/1.010/0.007       100,930,921
 2     35.523%      1.605/1.625/1.634/0.007        75,289,402
 3     34.202%      1.910/1.940/1.955/0.011        72,490,888
 4     32.935%      2.333/2.348/2.359/0.007        69,804,533
 5     32.667%      2.489/2.516/2.529/0.010        69,236,516
 6     32.515%      2.917/2.936/2.945/0.007        68,915,050
 7     32.257%      3.878/3.910/3.924/0.010        68,367,409
 8     32.172%      6.151/6.173/6.190/0.012        68,186,790
 9     32.156%      8.615/8.668/8.698/0.023        68,154,025

 avg1  34.672%                        3.457
 tot                                933.511       661,375,534
```
## https://github.com/zlib-ng/zlib-ng/pull/785/commits/7e31d5c4122584cbdfa4f7d60bba0341d3dec2d7
```
 Tool: minigzip-warn-x.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      0.995/1.007/1.012/0.005       100,930,921
 2     35.523%      1.602/1.622/1.633/0.009        75,289,402
 3     34.202%      1.928/1.944/1.956/0.008        72,490,888
 4     32.935%      2.325/2.344/2.357/0.009        69,804,533
 5     32.667%      2.486/2.512/2.523/0.008        69,236,516
 6     32.515%      2.900/2.930/2.940/0.008        68,915,050
 7     32.257%      3.898/3.913/3.922/0.007        68,367,409
 8     32.172%      6.134/6.170/6.193/0.014        68,186,790
 9     32.156%      8.636/8.685/8.719/0.027        68,154,025

 avg1  34.672%                        3.459
 tot                                933.859       661,375,534

 Tool: minigzip-warn-x.exe Levels: 1-9
 Runs: 70         Trim worst: 40

 Level    Comp  Comptime min/avg/max/stddev   Compressed size
 1     47.621%      0.995/1.007/1.013/0.004       100,930,921
 2     35.523%      1.595/1.622/1.632/0.009        75,289,402
 3     34.202%      1.915/1.948/1.962/0.012        72,490,888
 4     32.935%      2.331/2.348/2.359/0.009        69,804,533
 5     32.667%      2.503/2.515/2.526/0.007        69,236,516
 6     32.515%      2.909/2.933/2.943/0.009        68,915,050
 7     32.257%      3.896/3.922/3.936/0.011        68,367,409
 8     32.172%      6.134/6.169/6.192/0.014        68,186,790
 9     32.156%      8.603/8.673/8.707/0.028        68,154,025

 avg1  34.672%                        3.460
 tot                                934.128       661,375,534
```

**Comment by Dead2** [comment]
## Baseline 1da1491609c313eafbe6832d6cf024035f6261e6
```
   text    data     bss     dec     hex filename
 112377    1376      32  113785   1bc79 libz-ng.so.1

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      0.983/0.999/1.010/0.008        0.626/0.647/0.658/0.008      136,387,432
 2     43.875%      0.985/1.008/1.015/0.007        0.347/0.363/0.372/0.007       55,234,517
 3     42.490%      1.064/1.082/1.089/0.007        0.293/0.311/0.317/0.005       46,804,345
 4     41.468%      1.103/1.112/1.118/0.005        0.247/0.258/0.264/0.005       39,153,433
 5     41.212%      0.980/0.998/1.005/0.007        0.206/0.215/0.219/0.004       32,425,995
 6     41.037%      0.915/0.936/0.940/0.005        0.162/0.169/0.175/0.004       25,831,161
 7     40.784%      1.172/1.182/1.188/0.005        0.161/0.170/0.174/0.003       25,671,711
 8     40.707%      1.119/1.135/1.141/0.005        0.114/0.126/0.130/0.004       19,217,362
 9     40.696%      1.243/1.261/1.268/0.006        0.114/0.127/0.130/0.004       19,212,390

 avg1  43.339%                        1.079                          0.265
 tot                                291.356                         71.552      399,938,346

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      0.982/0.997/1.007/0.008        0.624/0.644/0.655/0.010      136,387,432
 2     43.875%      0.997/1.010/1.016/0.005        0.355/0.364/0.369/0.004       55,234,517
 3     42.490%      1.069/1.082/1.089/0.006        0.300/0.310/0.317/0.005       46,804,345
 4     41.468%      1.100/1.108/1.114/0.004        0.247/0.258/0.264/0.006       39,153,433
 5     41.212%      0.990/0.999/1.006/0.005        0.199/0.212/0.219/0.006       32,425,995
 6     41.037%      0.922/0.930/0.936/0.004        0.161/0.170/0.175/0.004       25,831,161
 7     40.784%      1.173/1.183/1.188/0.004        0.161/0.168/0.174/0.004       25,671,711
 8     40.707%      1.125/1.136/1.141/0.004        0.117/0.126/0.130/0.004       19,217,362
 9     40.696%      1.247/1.261/1.266/0.005        0.113/0.124/0.130/0.004       19,212,390

 avg1  43.339%                        1.079                          0.264
 tot                                291.227                         71.273      399,938,346
```

## PR #785 7e31d5c4122584cbdfa4f7d60bba0341d3dec2d7
```
   text    data     bss     dec     hex filename
 112441    1376      32  113849   1bcb9 libz-ng.so.1

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      0.981/1.000/1.009/0.007        0.608/0.640/0.648/0.009      136,387,432
 2     43.875%      0.989/1.005/1.012/0.006        0.354/0.364/0.371/0.005       55,234,517
 3     42.490%      1.067/1.081/1.088/0.006        0.300/0.312/0.320/0.006       46,804,345
 4     41.468%      1.094/1.111/1.119/0.007        0.245/0.258/0.264/0.005       39,153,433
 5     41.212%      0.989/1.000/1.005/0.004        0.205/0.212/0.216/0.003       32,425,995
 6     41.037%      0.921/0.932/0.939/0.006        0.158/0.170/0.175/0.004       25,831,161
 7     40.784%      1.167/1.179/1.185/0.005        0.151/0.168/0.174/0.005       25,671,711
 8     40.707%      1.127/1.134/1.139/0.003        0.111/0.126/0.130/0.004       19,217,362
 9     40.696%      1.246/1.259/1.265/0.005        0.110/0.126/0.130/0.005       19,212,390

 avg1  43.339%                        1.078                          0.264
 tot                                291.067                         71.303      399,938,346

 Level   Comp   Comptime min/avg/max/stddev  Decomptime min/avg/max/stddev  Compressed size
 1     57.780%      0.977/0.998/1.007/0.008        0.625/0.645/0.653/0.006      136,387,432
 2     43.875%      0.994/1.008/1.017/0.005        0.352/0.364/0.371/0.005       55,234,517
 3     42.490%      1.059/1.081/1.088/0.007        0.297/0.308/0.317/0.005       46,804,345
 4     41.468%      1.101/1.114/1.119/0.005        0.248/0.258/0.264/0.005       39,153,433
 5     41.212%      0.991/1.003/1.009/0.006        0.202/0.213/0.219/0.005       32,425,995
 6     41.037%      0.929/0.935/0.939/0.003        0.158/0.168/0.175/0.005       25,831,161
 7     40.784%      1.164/1.181/1.188/0.006        0.157/0.169/0.174/0.004       25,671,711
 8     40.707%      1.117/1.132/1.138/0.005        0.117/0.125/0.130/0.004       19,217,362
 9     40.696%      1.252/1.261/1.267/0.005        0.117/0.124/0.130/0.003       19,212,390

 avg1  43.339%                        1.079                          0.264
 tot                                291.449                         71.248      399,938,346
```

This is looking great now. 😄

**Pull request merged**